### PR TITLE
Updated to include streams

### PR DESCRIPTION
**Pull request by tjchaplin**
Issue #1 - Support streams.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -2,85 +2,198 @@
...
+                beforeEach(function(){
+                    csvtojsonStream = gulpCsvtojson({genjs:true});
+                });
+
+                it("should generate expected json string",function(done){
+
+                    var expectedResult = 'var any = [{"Header1":"col1-row1","Header2":"col2-row1","Header3":"col3-row1"},{"Header1":"col1-row2","Header2":"col2-row2","Header3":"col3-row2"}];';
+                    csvtojsonStream.on('data',function(file){
+                        file.contents.pipe(concat(function(data){
+                            assert(data.toString().replace(/\s\n/g,'') == expectedResult);
```
Comment (wizicer) [inline test.js]: Is it better to use `assert.equal` which produce rich result (output the expected and actual value) when failed?

File: test.js [hunk 2]
```diff
@@ -2,85 +2,198 @@
 var assert = require('assert');
 var gutil = require('gulp-util');
 var csvtojson = require('./');
```
Comment (wizicer) [inline test.js]: this line of code has been deprecated by your followed line. better remove it.

**Comment by tjchaplin** [comment]
Updated per comment.

Also, refactored the tests and index to allow for more readability.

**Comment by tjchaplin** [comment]
Further reviewing the code...it seems like there are too many paths for the plugin.  Ideally there should be a single purpose; i.e. stream `csvtojson` module using gulp.  The current plugin is streaming, but also will create a js file depending on options.  The creation of a JS file could(should) be done in a different plugin.  Here is the gulp plugin guidelines which provides a little more detail: [Gulp Plugin Guidelines](https://github.com/gulpjs/gulp/blob/master/docs/writing-a-plugin/guidelines.md)

I have opened up another issue(#3) to discuss further.

Here is what currently exists:

``` javascript
var gulp = require('gulp');
var csvtojson = require('gulp-csvtojson');

gulp.task('default', function () {
    return gulp.src('src/file.csv')
        .pipe(csvtojson())
        .pipe(gulp.dest('dist'));
});
```

I think this should be:

``` javascript
var gulp = require('gulp');
var csvtojson = require('gulp-csvtojson');
var insert = require('gulp-insert'); //module to insert into streamed data


gulp.task('default', function () {
    return gulp.src('src/file.csv')
        .pipe(csvtojson())
        .pipe(insert.prepend('var anyVariable = '))
        .pipe(insert.append(';'))
        .pipe(gulp.dest('dist'));
});
```

[Gulp Insert plugin](https://github.com/rschmukler/gulp-insert)

File: test/buffers/options/jsOptionsSpec.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	                    	var expected = "var any = " + JSON.stringify(csvData.expected) + ";";
+	                        assert.equal(streamedfile.contents.toString(), expected);
+	                        done();
+	                    });
+
+	                    csvtojsonStream.write(file);
+	                    csvtojsonStream.end();
+	                });
+				});
+	    		describe("When globalvariable is not set",function(){
```
Comment (wizicer) [inline test/buffers/options/jsOptionsSpec.js]: this case should be **When globalvariable is ~~not~~ set**

File: test/buffers/options/defaultOptionsSpec.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+		        it("should generate expected json string "+csvData.name,function(done){
+		            csvtojsonStream.once('data',function(streamedfile){
+		                assert.equal(streamedfile.contents.toString(), JSON.stringify(csvData.expected));
+		                done();
+		            });
+		            csvtojsonStream.write(file);
+		            csvtojsonStream.end();
+		        });
+		        it("should rename path to a json file"+csvData.name,function(done){
```
Comment (wizicer) [inline test/buffers/options/defaultOptionsSpec.js]: as `csvData.name` already appears in the upper level `describe`, no need extra one here.

**Comment by wizicer** [comment]
`+1` for separating fixtures from cases, however, obvious code duplication between `buffers` and `streams` directory which also produces problematic code twice.

Anyway, I suggest for this PR, let's just focus on this stream support, let's merge the PR after the two issues I just raise has been resolved. Then we can continue enhance with less code duplication later.

**Comment by tjchaplin** [comment]
Yes agree on some duplication...and we can review that.  I think if the streams are pushed in.  Maybe work on #3 , and remove tests/duplication from there.

Let me know if anything else.

**Pull request merged**